animareflection/ui
# Menu Component and ANIREF Preset Changes

## Description

##### Task link: https://trello.com/c/5vPb4Lpt/115-menu-component

Added a `Menu` component that supports basic usage, grouping elements, as well as extensions through submenus. Also went out of scope with this PR (my bad) and added theme toggle capability to storybook for testing purposes. The current implementation is not ideal because you must also manually switch the `iframe` background color to get the full effect. The benefit though is that we can now apply responsive styles and they will be rendered appropriately in storybook. With this addition, I also went and updated some details for our ANIREF preset, mainly surrounding `colors.ts` and newly added `semanticTokens.ts`.

## Test Steps

1) Verify that everything builds appropriately with newly added `Menu` component and preset changes.
2) Verify that stories presented in storybook render appropriately and look good.
3) Test functionality of theme switching in storybook (trigger is a `moon` or `sun` icon located in the toolbar above the main `iframe`). You will have to switch the background color in some cases to get full effect (trigger is also located in the toolbar above the main `iframe`).
4) Test overall appearance of all components, they should be responsive based on current theme.
5) Test functionality of `Menu` component in a downstream app.
